### PR TITLE
feat(testttp): extend testttp with new JSON methods and refactors test to use it

### PR DESCRIPTION
**Pull request by jsteenb2**
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: pkg/testttp/http.go [hunk 1]
```diff
@@ -10,76 +12,94 @@ import (
...
-		Req: req,
+		t:   r.t,
+		Req: r.req,
 		Rec: rec,
 	}
 }
 
+// Headers allows the user to set headers on the http request.
+func (r *Req) Headers(k, v string, rest ...string) *Req {
+	headers := append(rest, k, v)
```
Comment (kelwang) [inline pkg/testttp/http.go]: Why don't you use a map, looks like a little hacky.

Comment (jsteenb2) [reply pkg/testttp/http.go]: maps makes things unique, don't want to do that here. You can have multiple header values. This design is intentional.

Comment (kelwang) [reply pkg/testttp/http.go]: The key should unique, right?

Comment (jsteenb2) [reply pkg/testttp/http.go]: @kelwang sorry for the earlier reply, let that in haste, was rushing out the door. Here is better answer 😬.

When you are using this pkg in a test, the vast majority of hte time (in my experience), has been to add a single header. This design allows for that most common use to be super dead simple:

```go
testttp.
     Post(...).
     Header("Content-Type", "application/json").
      ...  more stuffs
```

if I wnat to add additional headers I still can, but I dont' ahve to sacrifice the ease of using this API to do that. I can simply do the following:

```go
testttp.
     Post(...).
     Header(
         "Content-Type", "application/json",
         "Accept", "text/yaml",
      ).
      ...  more stuffs
```

The signature is a pattern I picked up from Dave Cheney(I'll try and see if I can pull up that blog post he shared about this). It's a means of providing a type safe API with compile time errors when at least 1 arg is required and the rest are optional. Here's an example to illustrate:

```go
func (r *Req) Headers(headerPairs ...string) *Req {...}

// no compile time help here
testtp.Post(...).Headers()... other stuffs 

// k,v estaglish input pattern (meaning they are pairs of k,v)
func (r *Req) Headers(k, v string, rest ...string) *Req {...}

// compile time error here
testtp.Post(...).Headers()... other stuffs 

```

**Pull request merged**